src/ChannelServer/World/MonsterBuilder.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Melia.Shared.Const;
+using Melia.Shared.World;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Melia.Channel.World
+{
+    class MonsterBuilder
```

Comment:
That is highly debatable in my opinion. Personally I dislike factories/builders and I don't see the appeal. What you're doing here can be done with the actual object just as easy, and constructors were *invented* to initialize classes with already available information. It's like you're moving the constructor to its own class. Not only that, you're basically creating a copy of `Monster`, forcing us to maintain one class for the price of two.

I think this is terribly inefficient.